framework/src/seclink/seclink.c

```diff
@@ -174,6 +174,48 @@ int sl_deinit(sl_ctx hnd)
 	return 0;
 }
 
+#ifdef CONFIG_ARM_TRUSTZONE
+int sl_allocate(sl_ctx hnd, uint32_t ssize)
+{
+	SL_ENTER;
+	if (!hnd || ((struct _seclink_s_ *)hnd)->fd <= 0) {
+		SL_ERR(-1);
+		return -1;
+	}
+
+	struct _seclink_s_ *sl = (struct _seclink_s_ *)hnd;
+	struct seclink_comm_info info = {.priv = NULL, .s_size = ssize, .s_context = NULL};
+	struct seclink_req req = {.req_type.comm = &info, 0};
+	DiagPrintf("nits alloc\n");
+	SL_CALL(sl, SECLINKIOC_ALLOC, req);
+
+	if (req.res != HAL_SUCCESS) {
+		SL_LOG("fail to allocate");
+	}
+	return SECLINK_OK;
+}
+
+int sl_free(sl_ctx hnd, uint32_t tz_memory_id)
+{
+	SL_ENTER;
+	if (!hnd || ((struct _seclink_s_ *)hnd)->fd <= 0) {
+		SL_ERR(-1);
+		return -1;
```

Comment:
How about using `SECLINK_ERROR` instead of `-1`?

Reply:
done